firebase-firestore/src/main/java/com/google/firebase/firestore/core/FirestoreClient.java

```diff
@@ -233,15 +226,19 @@ private void initialize(Context context, User user, boolean usePersistence, long
               databaseInfo.getDatabaseId(),
               serializer,
               params);
-      lruDelegate = sqlitePersistence.getReferenceDelegate();
+      LruDelegate lruDelegate = sqlitePersistence.getReferenceDelegate();
+      gc = lruDelegate.getGarbageCollector();
       persistence = sqlitePersistence;
-      scheduleLruGarbageCollection();
     } else {
       persistence = MemoryPersistence.createEagerGcMemoryPersistence();
     }
 
     persistence.start();
     localStore = new LocalStore(persistence, user);
+    if (gc != null) {
+      lruScheduler = gc.new Scheduler(asyncQueue, localStore);
```

Comment:
This looks really strange to me. Can this be `new gc.Scheduler()`?  I am not sure if that would work.

Reply:
Er, not static. Because it's not a static class. But a helper method.